### PR TITLE
MGDOBR-1204: Split (API) versioned artifacts into sub-modules

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-1204](https://issues.redhat.com/browse/MGDOBR-1204)

I will review `pom.xml`'s and `application.properties` in a subsequent PR.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/gateways/Action.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.redhat.service.smartevents.infra.core.models.gateways;
```
Comment (r00ta) [inline infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/gateways/Action.java:1]: what about moving all the `gateways` to `v1`?

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/filters/BaseFilter.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.redhat.service.smartevents.infra.models.filters;
+package com.redhat.service.smartevents.infra.core.models.filters;
```
Comment (r00ta) [inline infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/filters/BaseFilter.java:1]: What about moving the filters to `v1`?

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/connectors/ConnectorType.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package com.redhat.service.smartevents.infra.models.connectors;
+package com.redhat.service.smartevents.infra.core.models.connectors;
```
Comment (r00ta) [inline infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/connectors/ConnectorType.java:1]: what about moving the connectors to `v1`?

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/processors/ProcessorType.java [hunk 4]
```diff
@@ -1,7 +1,7 @@
-package com.redhat.service.smartevents.infra.models.processors;
+package com.redhat.service.smartevents.infra.core.models.processors;
```
Comment (r00ta) [inline infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/processors/ProcessorType.java:1]: move to `v1`?

File: infra-core/pom.xml [hunk 5]
```diff
@@ -10,7 +10,7 @@
   </parent>
 
   <modelVersion>4.0.0</modelVersion>
-  <artifactId>infra</artifactId>
+  <artifactId>infra-core</artifactId>
```
Comment (r00ta) [inline infra-core/pom.xml:13]: what about creating an `infra` packaging-pom with 3 submodules `core`, `v1` and `v2`?

File: manager/pom.xml [hunk 6]
```diff
@@ -14,182 +14,17 @@
   <name>SmartEvents :: Manager</name>
 
   <dependencies>
-    <dependency>
-      <groupId>io.quarkus</groupId>
```
Comment (r00ta) [inline manager/pom.xml:18]: same here? A `manager` packaging-pom with `core`, `v1`, `v2` and `runner` as submodules?

File: manager-core/src/main/java/com/redhat/service/smartevents/manager/core/connectors/ConnectorsOidcClient.java [hunk 7]
```diff
@@ -9,8 +9,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.redhat.service.smartevents.infra.auth.AbstractOidcClient;
-import com.redhat.service.smartevents.infra.auth.OidcClientConfigUtils;
+import com.redhat.service.smartevents.infra.core.auth.AbstractOidcClient;
+import com.redhat.service.smartevents.infra.core.auth.OidcClientConfigUtils;
```
Comment (r00ta) [inline manager-core/src/main/java/com/redhat/service/smartevents/manager/core/connectors/ConnectorsOidcClient.java:13]: move `connectors` to `v1`?

**Review by r00ta — COMMENTED** [review]
Nice! Just a couple of questions :thumbs

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/gateways/Action.java (hunk 1 above)
Comment (manstis) [reply infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/gateways/Action.java:1]: Agreed. Done.

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/filters/BaseFilter.java (hunk 2 above)
Comment (manstis) [reply infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/filters/BaseFilter.java:1]: Agreed. Done.

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/connectors/ConnectorType.java (hunk 3 above)
Comment (manstis) [reply infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/connectors/ConnectorType.java:1]: Agreed. Done.

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/processors/ProcessorType.java (hunk 4 above)
Comment (manstis) [reply infra-core/src/main/java/com/redhat/service/smartevents/infra/core/models/processors/ProcessorType.java:1]: Agreed. Done.

File: infra-core/src/main/java/com/redhat/service/smartevents/infra/core/transformations/TransformationEvaluator.java [hunk 8]
```diff
@@ -1,4 +1,4 @@
-package com.redhat.service.smartevents.infra.transformations;
+package com.redhat.service.smartevents.infra.core.transformations;
```
[the remark this answers is not in the document]
Comment (manstis) [reply infra-core/src/main/java/com/redhat/service/smartevents/infra/core/transformations/TransformationEvaluator.java:1]: Agreed. Done.

File: manager-core/src/main/java/com/redhat/service/smartevents/manager/core/connectors/ConnectorsOidcClient.java (hunk 7 above)
Comment (manstis) [reply manager-core/src/main/java/com/redhat/service/smartevents/manager/core/connectors/ConnectorsOidcClient.java:13]: Agreed. Done.

**Review by r00ta — APPROVED** [review]
LGTM

**Review by lucamolteni — APPROVED** [review]

**Pull request merged**